### DISCUSSION TITLE
How to play created in code Audio File from code?

### DISCUSSION THREAD

**Discussion by Buran-company** [stride3d/stride#1644] [Q&A]
Hello everyone, I'm trying to implement this thing: create an audio file in script and play it.
So, I have .py file, where an audio file creates and saves in a directory (well, I have the path to the audio file and I should operate with this)
Then, I have to load this audio file in script, but it's a tricky one. 
I wanted to create Sound Asset in script, but I can't save it (serialization error), and I don't know, how to attach an audio file to Sound Asset in script.
In Stride documentation I saw information about creating custom audio data, I checked code sample, but I don't know, how to create CustomSoundSource without using SoundInstance.
Please, help me to resolve this question, it's a very important thing to me.

**Comment by tebjan** [comment]
Do you want to load an audio file in a script, or do you want to create the audio samples/data generatively in code?

**Reply by Buran-company** [reply]
Well, I just want to load an audio file in a script and play it

**Reply by tebjan** [reply]
And this file should open via a file browser? If not, why can't you add it as an asset as described in [the audio manual](https://doc.stride3d.net/latest/en/manual/audio/index.html)?

**Reply by Buran-company** [reply]
This audio file creates in a script too, so it should be done without using the Editor...
Because creating and loading an audio file are implementing while game is running

**Reply by manio143** [reply]
The Stride audio components currently are limited to load sounds from compiled assets. However, the actual system for playing sounds can be used outside of the component system. Less nice and you might not be able to get the 3d effects for example, but if you just want to play a sound you can look at this code of mine which plays a file in PCM format stereo https://github.com/manio143/BeatDetection/blob/master/BeatDetection.Tester/StrideAudio.cs
